plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -975,7 +975,9 @@ private <T> ThreadedActionListener<T> threadedActionListener(String threadPoolNa
      * @param runningTaskLimit limit
      */
     public void checkAndAddRunningTask(MLTask mlTask, Integer runningTaskLimit) {
-        if (Objects.nonNull(mlTask) && mlTask.getFunctionName() != FunctionName.REMOTE) {
+
+        // for agent and remote model prediction we don't need to check circuit breaker
+        if (Objects.nonNull(mlTask) && mlTask.getFunctionName() != FunctionName.REMOTE && mlTask.getFunctionName() != FunctionName.AGENT) {
```

Comment:
For agent using pre-trained model, does this skip the CB?

Reply:
That a good question. It will skip agent level execution. But under the agent execution when a model prediction is going to happen, then function name is not remote or agent, so that time it will check CB.

Reply:
Sure we can merge for now but need to remember that this would cause problems when agent is not using a remote model.

Reply:
For old versions, can user update setting to disable circuit breaker checking ?

Reply:
From [code](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/breaker/MLCircuitBreakerService.java#L78-L81), looks like we only check [memory](https://docs.opensearch.org/docs/latest/ml-commons-plugin/cluster-settings/#set-jvm-heap-memory-threshold) and [disk](https://docs.opensearch.org/docs/latest/ml-commons-plugin/cluster-settings/#set-jvm-heap-memory-threshold). 

So if somebody sets up, `jvm_heap_memory_threshold` to 100 & `disk_free_space_threshold` to -1 then circuit breaker will be disabled. 

Does that answer your question @ylwu-amzn ?

Reply:
Sounds good, @dhrubo-os , can you check if we have this in our document?

Reply:
which document are you referring to? I shared the memory and disk link from docs.opensearch.org.